packages/intrepid2/src/Cell/Intrepid2_CellDataDef.hpp

```diff
@@ -826,61 +826,61 @@ refCenterDataStatic_ = {
 // Point Inclusion 
 
 
-  template<typename PointViewType>
+  template<typename PointViewType, typename ScalarType>
   KOKKOS_INLINE_FUNCTION
   bool
   PointInclusion<shards::Line<>::key>::
-  check(const PointViewType &point, const double threshold) {
+  check(const PointViewType &point, const ScalarType threshold) {
     const double minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
     return (minus_one <= point(0) && point(0) <= plus_one);
   }  
 
-  template<typename PointViewType>
+  template<typename PointViewType, typename ScalarType>
   KOKKOS_INLINE_FUNCTION
   bool
   PointInclusion<shards::Triangle<>::key>::
-  check(const PointViewType &point, const double threshold) {
+  check(const PointViewType &point, const ScalarType threshold) {
     const double distance = max( max( -point(0), -point(1) ), point(0) + point(1) - 1.0 );
     return distance < threshold;
   }
   
-  template<typename PointViewType>
+  template<typename PointViewType, typename ScalarType>
   KOKKOS_INLINE_FUNCTION
   bool
   PointInclusion<shards::Quadrilateral<>::key>::
   check(const PointViewType &point, 
-                      const double threshold) {
+                      const ScalarType threshold) {
     const double minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
     return ((minus_one <= point(0) && point(0) <= plus_one) &&
             (minus_one <= point(1) && point(1) <= plus_one));
   }  
 
-  template<typename PointViewType>
+  template<typename PointViewType, typename ScalarType>
   KOKKOS_INLINE_FUNCTION
   bool
   PointInclusion<shards::Tetrahedron<>::key>::
-  check(const PointViewType &point, const double threshold) {
+  check(const PointViewType &point, const ScalarType threshold) {
     const double distance = max( max(-point(0),-point(1)),
                                   max(-point(2), point(0) + point(1) + point(2) - 1) );
     return distance < threshold;
   }
 
-  template<typename PointViewType>
+  template<typename PointViewType, typename ScalarType>
   KOKKOS_INLINE_FUNCTION
   bool
   PointInclusion<shards::Hexahedron<>::key>::
-  check(const PointViewType &point, const double threshold) {
+  check(const PointViewType &point, const ScalarType threshold) {
     const double minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
     return ((minus_one <= point(0) && point(0) <= plus_one) &&
             (minus_one <= point(1) && point(1) <= plus_one) &&
             (minus_one <= point(2) && point(2) <= plus_one));
   }
   
-  template<typename PointViewType>
+  template<typename PointViewType, typename ScalarType>
   KOKKOS_INLINE_FUNCTION
   bool
   PointInclusion<shards::Pyramid<>::key>::
-  check(const PointViewType &point, const double threshold) {
+  check(const PointViewType &point, const ScalarType threshold) {
     const double minus_one = -1.0 - threshold, plus_one = 1.0 + threshold, minus_zero = -threshold;
```

Comment:
Should `minus_one`, etc. have type `auto` or `ScalarType` instead?  (If the only other case we're supporting is `float`, it will be fine as it is, maybe modulo some compiler warnings for the comparison below, but if we intend it to be more general, I expect `auto` or `ScalarType` is the right thing, here and throughout.)

Reply:
Good point! I pushed a fix using ScalarType instead of double.